components/script_injector/renderer/script_injector_render_frame_observer.cc

```diff
@@ -0,0 +1,70 @@
+// Copyright (c) 2023 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this file,
+// You can obtain one at https://mozilla.org/MPL/2.0/.
+
+#include "brave/components/script_injector/renderer/script_injector_render_frame_observer.h"
+
+#include <string>
+#include <utility>
+
+#include "content/public/renderer/render_frame.h"
+#include "third_party/blink/public/common/associated_interfaces/associated_interface_registry.h"
+#include "third_party/blink/public/web/web_local_frame.h"
+#include "third_party/blink/public/web/web_script_source.h"
+
+namespace script_injector {
+
+ScriptInjectorRenderFrameObserver::ScriptInjectorRenderFrameObserver(
+    content::RenderFrame* render_frame)
+    : content::RenderFrameObserver(render_frame) {
+  render_frame->GetAssociatedInterfaceRegistry()
+      ->AddInterface<script_injector::mojom::ScriptInjector>(
+          base::BindRepeating(&ScriptInjectorRenderFrameObserver::BindReceiver,
+                              base::Unretained(this)));
+}
+
+void ScriptInjectorRenderFrameObserver::OnDestruct() {
+  delete this;
+}
+
+void ScriptInjectorRenderFrameObserver::BindReceiver(
+    mojo::PendingAssociatedReceiver<script_injector::mojom::ScriptInjector>
+        pending_receiver) {
+  receiver_.reset();
```

Comment:
there could be more than one receiver at a time. The method name convention is `OnScriptInjectorRenderFrameObserverRequest` and you should use `mojo::AssociatedReceiverSet`

Reply:
Where are you seeing that convention? Even with multiple receivers, I would just want to add them one at a time: https://source.chromium.org/chromium/chromium/src/+/main:components/on_load_script_injector/renderer/on_load_script_injector.cc;l=28;bpv=1;bpt=1